### PR TITLE
Collect TCP statistics

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

File: subsys/net/ip/connection.c [hunk 1]
```diff
@@ -871,11 +873,17 @@ enum net_verdict net_conn_input(enum net_ip_protocol proto, struct net_pkt *pkt)
...
+			net_stats_update_tcp_seg_connrst();
+		}
 	}
 
 drop:
 	if (proto == IPPROTO_UDP) {
 		net_stats_update_udp_drop();
+	} else if (IS_ENABLED(CONFIG_NET_TCP) && proto == IPPROTO_TCP) {
+		net_stats_update_tcp_seg_drop();
 	}
```
Comment (tbursztyka) [inline subsys/net/ip/connection.c]: Hum, why not wrapping this if/else if (as well as the update recv above) in an inline function such as: 
net_stats_update_per_proto_recv(proto);
and:
net_stats_update_per_proto_drop(proto);

**Review by tbursztyka — CHANGES_REQUESTED** [review]

File: subsys/net/ip/connection.c (hunk 1 above)
Comment (lpereira) [inline subsys/net/ip/connection.c]: +1 for this

Comment (jukkar) [inline subsys/net/ip/connection.c]: Sure, why not. I will cook a new version.

**Comment by jukkar** [comment]
New version sent. Added also new patch that adds possibility to update UDP checksum error count.

**Comment by jukkar** [comment]
Added two more statistics related fixes that I noticed while testing the statistics printing.

File: include/net/net_stats.h [hunk 2]
```diff
@@ -90,22 +101,25 @@ struct net_stats_tcp {
 	/** Number of TCP segments with a bad checksum. */
 	net_stats_t chkerr;
 
-	/** Number of TCP segments with a bad ACK number. */
+	/** Number of received TCP segments with a bad ACK number. */
 	net_stats_t ackerr;
 
+	/** Number of received TCP bad RST (reset) segments. */
```
Comment (pfalcon) [inline include/net/net_stats.h]: Minor nitpick: "TCP bad RST" sounds strange, maybe "bad TCP RST"?

**Review by pfalcon — APPROVED** [review]
I'd also find enabling stats by default strange. As a further step on https://jira.zephyrproject.org/browse/ZEP-2105, I wanted to propose to enable NET_ERR logging by default (so the case of no netif's was visibly reported to a user). Assuming we can only have one enabled by default - stats or err logging, I'd prefer logging.

**Comment by jukkar** [comment]
> I'd also find enabling stats by default strange.

Please note that we enable only statistics for given subsystem like RPL if both CONFIG_NET_STATISTICS and CONFIG_NET_RPL is defined. IMHO this is quite logical now so that user does not have to separately enable RPL statistics in this case.

File: include/net/net_stats.h (hunk 2 above)
Comment (jukkar) [inline include/net/net_stats.h]: sure, I will fix that

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**